app/controllers/media_controller.rb

```diff
@@ -0,0 +1,24 @@
+require 'will_paginate/array'
+
+class MediaController < ApplicationController
+  def index
+    if !current_spy.nil?
```

Comment:
This would be a great place to use the ```logged_in?``` helper method you defined in the ApplicationController.